In [90]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from sklearn.model_selection import train_test_split
SEED = 2000

In [93]:
data = pd.read_csv("../data/clean_tweet.csv", index_col=0)
data.head()

text  target
0  awww that bummer you shoulda got david carr of...       0
1  is upset that he can not update his facebook b...       0
2  dived many times for the ball managed to save ...       0
3     my whole body feels itchy and like its on fire       0
4  no it not behaving at all mad why am here beca...       0

In [94]:
x = data.text
y = data.target

In [95]:
x_train, x_validation_test, y_train, y_validation_test = train_test_split(x,y, test_size=0.02, random_state=SEED)

In [96]:
x_train.shape

(1564120,)

In [97]:
x_validation_test.shape

(31921,)

In [98]:
x_validation, x_test, y_validation, y_test = train_test_split(x_validation_test, y_validation_test, test_size=0.5, random_state=SEED)

In [99]:
print(f"Train set has total {len(x_train)} with {len(x_train[y_train==0])*100/(len(x_train)*1.0)}% negative and {len(x_train[y_train==4])*100/(len(x_train)*1.0)}% positive")

Train set has total 1564120 with 50.020139119760636% negative and 0.0% positive


In [100]:
print(f"Validation set has total {len(x_validation)} with {len(x_validation[y_validation==0])*100/(len(x_validation)*1.0)}% negative and {len(x_validation[y_validation==4])*100/(len(x_validation)*1.0)}% positive")

Validation set has total 15960 with 49.454887218045116% negative and 0.0% positive


In [101]:
print(f"Test set has total {len(x_test)} with {len(x_test[y_test==0])*100/(len(x_test)*1.0)}% negative and {len(x_test[y_test==4])*100/(len(x_test)*1.0)}% positive")

Test set has total 15961 with 49.67733851262452% negative and 0.0% positive


### Baseline

The basline can be ZeroRule classifier (classifiy the majority class always). That would give an accuracy of 50.02%.

We are using the inbuild sentiment analysis available in `TextBlob` library.

In [102]:
from textblob import TextBlob
from sklearn.metrics import accuracy_score, classification_report, confusion_matrix

In [103]:
textblob_result = [TextBlob(i).sentiment_assessments for i in x_validation]

In [104]:
textblob_result[0]

Sentiment(polarity=0.0, subjectivity=0.06666666666666667, assessments=[(['last'], 0.0, 0.06666666666666667, None)])

In [105]:
textblob_predictions = [0 if n.polarity < 0 else 1 for n in textblob_result]

In [106]:
type(y_validation)

pandas.core.series.Series

In [118]:
confusion_mat = confusion_matrix(y_validation, textblob_predictions, labels=[1,0])
confusion_df = pd.DataFrame(confusion_mat, index=['positive', 'negative'], columns=['predicted_positive', 'predicted_negative'])

In [119]:
print(f"Accuracy score is {accuracy_score(y_validation, textblob_predictions)*100.0}")
print("-"*80)
print("Confusion Matrix\n")
print(confusion_df)
print("-"*80)
print("Classification Report\n")
print(classification_report(y_validation, textblob_predictions))

Accuracy score is 61.83583959899749
--------------------------------------------------------------------------------
Confusion Matrix

          predicted_positive  predicted_negative
positive                7282                 785
negative                5306                2587
--------------------------------------------------------------------------------
Classification Report

              precision    recall  f1-score   support

           0       0.77      0.33      0.46      7893
           1       0.58      0.90      0.71      8067

    accuracy                           0.62     15960
   macro avg       0.67      0.62      0.58     15960
weighted avg       0.67      0.62      0.58     15960

